tools/RAiDER/downloadGNSSDelays.py

```diff
@@ -166,18 +172,22 @@ def download_tropo_delays(stats, years, writeDir='.', download=False, verbose=Fa
     # Iterate over stations and years and check or download data
     results = []
     stat_year_tup = itertools.product(stats, years)
-    for s, y in itertools.product(stats, years):
-        if verbose:
-            print('Currently checking station {} in {}'.format(s, y))
-        fileurl = download_UNR(s, y, writeDir=writeDir,
-                               download=download, verbose=verbose)
-        # only record valid path
-        if fileurl:
-            results.append({'ID': s, 'year': y, 'path': fileurl})
-    return pd.DataFrame(results).set_index('ID')
+    stat_year_tup = (tup + (writeDir,) + (download,) + (verbose,) for tup in stat_year_tup)
+    # Parallelize remote querying of station locations
+    with multiprocessing.Pool(numCPUs) as multipool:
+        for fileurl in multipool.starmap(download_UNR,stat_year_tup):
```

Comment:
BTW this whole thing could become a single list comprehension if you want
```suggestion
        for fileurl in multipool.starmap(download_UNR, stat_year_tup):
```